### PR TITLE
HACK: fixes pd.read_csv close issue

### LINKED ISSUE #1435: CI appears broken
Two separate PRs, test output [here](https://travis-ci.org/biocore/scikit-bio/jobs/165854392) and [here](https://travis-ci.org/biocore/scikit-bio/builds/165854390), impact is on both traces through build matrix. Failure appears to stem from repeated exceptions along the lines of:

```python
======================================================================
ERROR: test_custom_valid_multi_line (skbio.io.format.tests.test_blast6.TestBlast6Reader)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda3/envs/env_name/lib/python3.5/site-packages/skbio/io/format/tests/test_blast6.py", line 57, in test_custom_valid_multi_line
    'sallacc', 'qaccver'])
  File "/home/travis/miniconda3/envs/env_name/lib/python3.5/site-packages/skbio/io/registry.py", line 998, in wrapped_reader
    return reader_function(fhs[-1], **kwargs)
  File "/home/travis/miniconda3/envs/env_name/lib/python3.5/site-packages/skbio/io/format/blast6.py", line 272, in _blast6_to_data_frame
    "Specified number of columns (%r) does not equal"
  File "/home/travis/miniconda3/envs/env_name/lib/python3.5/site-packages/skbio/io/format/_blast.py", line 40, in _parse_blast_data
    fh.seek(0)
ValueError: I/O operation on closed file.
```

I've marked this as blocking as it blocks review on pull requests.

tagging #1352 and #1430

### PR DESCRIPTION
**Pull request by ebolyen**
Pinging @jairideout for later (he is on vacation atm): do you think the registry should be doing something like this anyways? It knows the appropriate lifetimes for these objects and it could neuter `close` until it was actually time to close.

Partially fixes #1435 

@RNAer, @wasade, or anyone else interested, would you be able to review/merge? Thanks!

### CODE REVIEW

**Comment by ebolyen** [comment]
Tests won't pass until #1436 is fixed as well.

File: skbio/io/format/_blast.py [hunk 1]
```diff
@@ -32,10 +33,28 @@ def _parse_blast_data(fh, columns, error, error_message, comment=None,
...
+            raise error(error_message % (len(columns), len(lineone.columns)))
+
+        fh.seek(0)
+
+        return read_csv(fh, names=columns, dtype=_possible_columns)
+
+
+# HACK for https://github.com/pandas-dev/pandas/issues/14418
+@contextlib.contextmanager
+def noop_close(fh):
```
Comment (wasade) [inline skbio/io/format/_blast.py]: Will defer as to whether this should be paired with a unit test. It's straight forward enough and exercised broadly by the test framework implicitly, so am okay without one.

Comment (ebolyen) [inline skbio/io/format/_blast.py]: I'll make private, thanks

**Comment by ebolyen** [comment]
Turns out you can set to `engine='python'` and there's no issue.

**Comment by ebolyen** [comment]
Actually it turns out `engine` and `dtype` cannot be used together in `read_csv`, so I guess this is still the way to go...

**Comment by wasade** [comment]
huh...

**Comment by jairideout** [comment]
@ebolyen looks good to me, probably worth having in the registry to match how standard Python I/O behaves (seems to be what ppl are bringing up in the pandas issue). Not high priority. I won't merge in case others are still reviewing

**Comment by ebolyen** [comment]
Thanks for the reviews all!

**Pull request merged**